src/symbols/MiniMap.tsx

```diff
@@ -0,0 +1,165 @@
+import React, { useMemo } from 'react';
+import { useStore } from '../store';
+
+interface MiniMapProps {
+  width?: number;
+  height?: number;
+  position?: 'top-left' | 'top-right' | 'bottom-left' | 'bottom-right';
+}
+
+export const MiniMap: React.FC<MiniMapProps> = ({
+  width = 200,
+  height = 150,
+  position = 'bottom-right'
+}) => {
+  // Get nodes, edges and theme from the store
+  const storeNodes = useStore(state => state.nodes);
+  const storeEdges = useStore(state => state.edges);
+  const theme = useStore(state => state.theme);
+  const actives = useStore(state => state.actives);
+  const selectedNodes = useStore(state => state.selections);
+  const hoveredNodeId = useStore(state => state.hoveredNodeId);
+
+  // Calculate positions based on actual node positions from the store
+  const nodePositions = useMemo(() => {
```

Comment:
Can you explain this more - its not super obvious whats going on here.

Reply:
Also if this really needs to stay, i would say probably needs to be extracted and unit tested.